config/config.go

```diff
@@ -674,6 +682,15 @@ type QueueConfig struct {
 	MaxBackoff model.Duration `yaml:"max_backoff,omitempty"`
 }
 
+// MetadataConfig is the configuration for sending metadata to remote
+// storage.
+type MetadataConfig struct {
+	// Send controls whenever we send metric metadata to remote storage.
+	Send bool `yaml:"send"`
+	// MetadataSendInterval controls how frequently we send metric metadata.
+	SendInterval model.Duration `yaml:"send_interval"`
```

Comment:
This name is confusing. As far as I can tell this value is used to control how often we gather metadata via the `collect` function`, which essentially controls how often we send metadata since we only append metadata to the queue manager from there. But from the name I would expect this directly controls how often we send, and wouldn't be sure which to modify to send more often given that flush deadline is also a hard "send data now".

Reply:
Great shout, given this is now hardcoded I have added a comment that reflects this - let me know what you think.